packages/rsocket-core/src/ClientServerMultiplexerDemultiplexer.ts

```diff
@@ -356,15 +483,31 @@ class SetupFrameHandler implements FlowControlledFrameHandler {
         )
         .then(
           (responder) => {
+            if (this.done) {
+              responder.close?.call(responder, this.error);
+              return;
+            }
+
             this.multiplexer["delegateHandler"] = new GenericFrameHandler(
```

Comment:
nit (for myself to come back to): property access of "delegateHandler" should not be allowed. Replacing the handler should be done through an encapsulating method.

Reply:
this should work since this is all files private and should not be that much important.